### PR TITLE
Add timezone support to JSON reader

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-array/src/types.rs [hunk 1]
```diff
@@ -287,30 +287,28 @@ impl ArrowTemporalType for DurationMicrosecondType {}
 impl ArrowTemporalType for DurationNanosecondType {}
 
 /// A timestamp type allows us to create array builders that take a timestamp.
-pub trait ArrowTimestampType: ArrowTemporalType {
+pub trait ArrowTimestampType: ArrowTemporalType<Native = i64> {
```
Comment (tustvold) [inline arrow-array/src/types.rs:290]: This is not a breaking change as these types are sealed, but it simplifies generic code as i64 is guaranteed

Comment (alamb) [reply arrow-array/src/types.rs:290]: I don't understand how these are sealed -- https://docs.rs/arrow/35.0.0/arrow/datatypes/trait.ArrowTimestampType.html

Maybe we should add a doc commen note here explaining that they are sealed

**Review by alamb — APPROVED** [review]

File: arrow-array/src/types.rs (hunk 1 above)
Comment (tustvold) [reply arrow-array/src/types.rs:290]: Oh, I assumed they were, they definitely should be :smile:

Comment (tustvold) [reply arrow-array/src/types.rs:290]: https://github.com/apache/arrow-rs/pull/3882

**Pull request merged**